inc/class-registry.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Component registry.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Components;
+
+/**
+ * Get an instance of the registry class.
+ *
+ * @return \WP_Irving\Components\Registry
+ */
+function get_registry() {
+	static $registry;
+
+	if ( empty( $registry ) ) {
+		$registry = new Registry();
+	}
+
+	return $registry;
+}
+
+/**
+ * Auto load some components which will then get registered. Defaults to the
+ * theme directory and WP Irving plugin.
+ */
+function auto_register_components() {
+
+	/**
+	 * Filter the directories from which components should be autoloaded.
+	 *
+	 * @param array $directories Array of directories.
+	 */
+	$directories = apply_filters(
+		'wp_irving_component_registry_directories',
+		[
+			'wp_irving' => WP_IRVING_PATH . '/inc/components/', // Load components from WP Irving.
+			'theme'     => get_stylesheet_directory() . '/components/', // Load components from the activated theme.
+		]
+	);
+
+	get_registry()->load_components( $directories );
+}
+add_filter( 'init', __NAMESPACE__ . '\auto_register_components' );
```

Comment:
Not a blocker, but I wonder if we should add our own `irving_init` action hook so that these types of things only get loaded if Irving is doing it's thing?

Reply:
Components could have other functionality that's required outside of an endpoint context, so I'd like to handle this separately from this PR with some additional consideration/discussion.